### PR TITLE
#57 death upon impact

### LINKED ISSUE #57: Death upon impact
**What is the use of the feature?**
Death upon impact would help make the game more challenging and  fun (to those who wish to turn it on.) It also adds more content; if people get bored of certain modes, they can try to master this. In the future. (Over 80% of people wanted to add this feature, which likely means it would get decent use)

**Describe the solution you'd like**
If your character crashes into an object past a certain speed (should be decently fast) the character will die. Needs to be optional.

### PR DESCRIPTION
**Pull request by AdamVromman**

### CODE REVIEW

**Comment by AdamVromman** [comment]
changing the impact threshold ingame isnt working at the moment, but I'm not sure if that's because of my code or because of an other bug

File: Assets/Scripts/Characters/Human/Hero.cs [hunk 1]
```diff
@@ -747,6 +747,19 @@ public void checkTitan()
         this.myTitans = list2;
     }
 
+    private void OnCollisionEnter(Collision collision)
+    {
+        
+        float force = collision.impulse.magnitude / Time.fixedDeltaTime;
```
Comment (Lithrun) [inline Assets/Scripts/Characters/Human/Hero.cs]: Why should the force be divided by Time.fixedDeltaTime? This value is always 0.02.

File: Assets/Scripts/Characters/Human/Hero.cs [hunk 2]
```diff
@@ -747,6 +747,19 @@ public void checkTitan()
...
+        if (FengGameManagerMKII.Gamemode.Settings.ImpactForce > 0 && force >= FengGameManagerMKII.Gamemode.Settings.ImpactForce)
+        {
+            die(new Vector3(), false);
+            FengGameManagerMKII.print(FengGameManagerMKII.Gamemode.Settings.ImpactForce);
```
Comment (Lithrun) [inline Assets/Scripts/Characters/Human/Hero.cs]: Print messages (Debug log) should not be included in a PR

**Review by Lithrun — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Characters/Human/Hero.cs (hunk 1 above)
Comment (AdamVromman) [reply Assets/Scripts/Characters/Human/Hero.cs]: This was included in the paper linked in the original thread (https://www.malgol.com/how-to-get-the-impact-force-of-a-collision-in-unity/). I dont understand the physics described, but if needed, i'll investigate some more.

**Review by Lithrun — APPROVED** [review]

**Pull request merged**